### ISSUE TITLE
Add mAP metric for object detection

### ISSUE THREAD

**Issue by TheCodez** [pytorch/ignite#520]
It would be a good idea to add a mAP metric which is often used in object detection. For this the newly added [IoU](https://github.com/pytorch/vision/blob/master/torchvision/ops/boxes.py#L102-L124) function from torchvision could be used.

**Comment by vfdev-5** [comment]
@TheCodez yes, that's can be a good idea. If you would like to contribute, it would be 👍

**Comment by TheCodez** [comment]
@vfdev-5 I have some code [here](https://github.com/TheCodez/pytorch-box2pix/blob/jit/metrics/mean_ap.py). It's probably not even working but maybe it can be used as a starting point. I don't think I can get this finished so it would be awesome if someone else could contribute here :)

**Comment by vfdev-5** [comment]
Few implementation refs: 
- https://github.com/coz-a/signate-fish-school/blob/main/src/solution/metrics.py
- https://github.com/bes-dev/mean_average_precision/

**Comment by aniketpandey05** [comment]
hey , @vfdev-5 I saw that  ignite/metrics/vision/object_detection_average_precision_recall.py  covers this issue already , is there anything specific that is pending?

**Comment by vfdev-5** [comment]
This should be closed, thanks!
